internal/engine/compiler/engine.go

```diff
@@ -39,6 +39,10 @@ type (
 		// name is the name the module was instantiated with used for error handling.
 		name string
 
+		// NOTE: The offset of the functions field is referenced from assembly based on
```

Comment:
so this is not moduleEngine specific thing but rather general thing about all the structs in this file.. so move this comment to https://github.com/tetratelabs/wazero/blob/main/internal/engine/compiler/engine.go#L25 ?

Reply:
basically like "be careful when you modify fields in this file!"

Reply:
Interesting - I think if the comment isn't exactly where a problematic field would be added, it wouldn't have prevented my lengthyish debugging session 😅 But it makes sense so moved it up

Reply:
yeah understood - basically we should dup the same comment across all the fields

Reply:
I tried adding some lightweight dup comments to increase the chance of safety without getting overblown, how does it look?